### ISSUE TITLE
Scaffolding Issue with .NET6.0 and Customized Interceptor

### ISSUE THREAD

**Issue by escwipe** [linq2db/linq2db#3820]
### Describe your issue

Dear LINQ TO DB Dev Team,

I have issue when using linq2db CLI scaffolding with ***.NET6.0*** with customized interceptor entity.  

I created a new class which is a derivative of *ScaffoldInterceptors* and inserted my source code (see section below *'Interceptor Class'*). In ***cli.json*** file, in the property **[general]**, I added the key **[customize]** with value of built **DLL**. 

I use **net6.0, linq2db 4.3.0 and linq2db.Tools 4.3.0** and when scaffolding, it shows an error. In your documentation you wrote that **customize** option, can be used on *.NET 3.1* or later versions and that it should be compatible with current runtime. After I changed project's target framework on netcoreapp3.1 and JSON **customize** option's value to new built in DLL *".\\bin\\Debug\\netcoreapp3.1\\PROJ.dll"*, everything works smooth and without errors.

### Summary:
When I use the framework .NET6.0 and scaffolding with customized interceptor entity. It fails.
When I use the framework .NET3.1 and scaffolding with customized interceptor entity. Everything runs perfectly fine.

Am I doing something wrong? What is the problem with customized scaffolding in the .NET6.0 environment?

### Steps to reproduce
```
dotnet linq2db scaffold --import "cli.json"
```

### cli.json
```
"general": {
  "output": ".\\Models",
  "overwrite": true,
  "provider": "SQLServer",
  "connection": "<< YOUR DB CONN STRING HERE >>",
  "template": "default",
  "customize": ".\\bin\\Debug\\net6.0\\PROJ.dll"
}
```

### Interceptor Class
```cs
public class Intreceptor : ScaffoldInterceptors
{
    // several use-cases cases of entity customization
    public override void PreprocessEntity(ITypeParser typeParser, EntityModel entityModel)
    {
        // for table with name "alltypes" we cannot recognize separate words to properly generate class name
        // let's modify generated entity class name in model
        if (entityModel.Class.Name.StartsWith("AspNet"))
            entityModel.Class.Name = entityModel.Class.Name.Replace("AspNet", String.Empty);

        // mark column as non-editable
        var creatorColumn = entityModel.Columns.Where(c => c.Metadata.Name == "created_by").FirstOrDefault();
        if (creatorColumn != null)
            creatorColumn.Metadata.SkipOnUpdate = true;
    }
  }
```

### Error
```
Unhandled exception: Unable to load one or more of the requested types.  
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
...
at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeModule.GetTypes()
   at System.Reflection.Assembly.GetTypes()
   at LinqToDB.CommandLine.ScaffoldCommand.LoadInterceptorsFromAssembly(String sourcePath, Assembly assembly, ScaffoldOptions options)
   at LinqToDB.CommandLine.ScaffoldCommand.LoadInterceptorsFromAssembly(String assemblyPath, ScaffoldOptions options)
   at LinqToDB.CommandLine.ScaffoldCommand.LoadInterceptors(String interceptorsPath, ScaffoldOptions options)
   at LinqToDB.CommandLine.ScaffoldCommand.Execute(CliController controller, String[] rawArgs, Dictionary`2 options, IReadOnlyCollection`1 unknownArgs)
   at LinqToDB.CommandLine.CliController.Execute(String[] args)
   at LinqToDB.Tools.Program.Main(String[] args)
```

### Environment details

Packages:
`Linq To DB` version: *4.3.0*
`Linq To DB Tools` version: *4.3.0*

Database (with version): *SQL Server 2019*

.NET Version: *6.0*

**Comment by MaceWindu** [comment]
We should update documentation. tool runs using .net core 3.1 runtime and cannot use assemblies built for newer runtimes

**Comment by zharonar** [comment]
> We should update documentation. tool runs using .net core 3.1 runtime and cannot use assemblies built for newer runtimes

Is there any info when or will it be updated for the newer runtimes?

**Comment by MaceWindu** [comment]
Is it really needed?

**Comment by zharonar** [comment]
![image](https://user-images.githubusercontent.com/28241734/199737857-492f807a-e737-4439-bbc8-22d0aa20b303.png)

And that million of times, which means, we cannot use it on newer frameworks without losing C# 10+ features etc... and global using directives if we use <ImplicitUsings> in csproj...

**Comment by MaceWindu** [comment]
It has nothing to do with TFM, to use specific language version you need to specify it in your csproj https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version (you can even use C# 11 with .net framework except couple of features, that depend on runtime).

We can add additional TFM builds for .net tool but it will result:
- bigger package size. Currently it is already 20Mb for one framework and if we add net6 +net7 it will be around 60Mb
- even more confusion: user need to specify which tfm to install during tool installation or he will get some default for current machine

I would say we can probably bump TFM to .net6 after .net 3.1 reach EOL which still will not solve issue if somebody try to build itnerceptors using net7.

**Comment by MaceWindu** [comment]
BTW, you can try to re-install tool with framework flag (`--framework net6.0`) https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-install#options

I didn't tested it yet but `linq2db.cli` were configured to run on newer frameworks, so it could work

**Comment by croban** [comment]
> BTW, you can try to re-install tool with framework flag (--framework net6.0)

This did not work for us :( 

Workaround for this issue is to create Interceptors inside another project (for netcoreapp3.1).
But because we use some base classes btw. interfaces on the generated models, we also have to move those infra. files to another shared project (for netcoreapp3.1 and net6.0) and clean up code - only using latest possible C# version - in this case that is C# 8.0. 

Another Workaround is stick with good old "obsolet" T4 lib.

**Comment by MaceWindu** [comment]
This sounds like a valid reason to add more TFMs. Will be done in next release

**Comment by MaceWindu** [comment]
> only using latest possible C# version - in this case that is C# 8.0.

As I mentioned above - this is not true. You can use any language version with any TFM

**Comment by viceroypenguin** [comment]
BTW if you are missing framework support for modern language features (init requires an attribute, as an example), you can reference the PolySharp library to poly-fill the necessary attributes to use latest C# with older TFMs.

**Comment by croban** [comment]
@MaceWindu THX, 
we have tried with cloning the project and we have tried to contribute, just we have had some problems with building it for the first time :(.  

About multiple TFMs, I did not find solution how to use e.g. File Scoped Namespaces and Global Usings within a project which is targeting multiple TFMs. That's why I am assuming that you have to stick with language features which are supported within smallest TFM for that project/library, or you could do checks with #if NETSTANDARD2_0 etc. 
But maybe I am wrong with this assumption. 

For us that would be huge overhead, because we don't make libs it is simple web app with exact one targeting framework and only thing what we would like is to generate DB model with linq2db.cli. where interceptors are the part of our datalayer project.

@viceroypenguin thx for the tipp - good to know

**Comment by viceroypenguin** [comment]
I use File Scoped Namespaces and Global Usings with C# 10 and targeting .netstandard 2.0, .net 3.1, .net 5.0, and .net 6.0 just fine. Mainly, what you need to have is:
```xml
  <PropertyGroup>
    <LangVersion>10</LangVersion>
    <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
  </PropertyGroup>
```

The `<LangVersion/>` tag will allow you to specify which version of C#, whether `latest` or `10`. The `<TargetFrameworks/>` tag will let you specify which TFMs you want to reference. Once you've done this, most C#10 features will be available natively, for both 3.1 and 6.0. This includes file-scoped namespaces, global usings, etc. If you want to use implicit usings, you can add `<ImplicitUsings>enable</ImplicitUsings>` as well.

However, as stated, some features require attributes in order to build correctly; you can find a full list on the [PolySharp](https://github.com/Sergio0694/PolySharp) GH page of which features these are and the attributes necessary. The easiest way to address this is to use that library to polyfill the necessary attributes - then you can use all C#10 features for any .netstandard, .netcore, or .net TFM.

**Comment by croban** [comment]
@viceroypenguin you are the KING 🥇 I will try that for sure!
